federation/schema.go

```diff
@@ -66,13 +67,55 @@ func validateFederationKeys(serviceNames []string, serviceSchemasByName map[stri
 						}
 					}
 				}
+				if isRootObject && !validFederatedKey {
+					return oops.Errorf("Invalid federation key %s", keyField)
+				}
+			}
+		}
+	}
+	return nil
+}
 
+// validateFederatedObjects validates that if a object is federated, it is federated on all the schemas
+func validateFederatedObjects(serviceNames []string, serviceSchemasByName map[string]*IntrospectionQueryResult, objName string) error {
+	// Check if it is federated on one service. It is federated if there is a field
+	//	called "__federation" on that object on any of the services it is on
+	federatedOnOneService := false
+	for _, service := range serviceNames {
+		for _, typ := range serviceSchemasByName[service].Schema.Types {
+			if typ.Name == objName {
+				for _, introspectedField := range typ.Fields {
+					if introspectedField.Name == federationField {
+						federatedOnOneService = true
+						break
+					}
+				}
 			}
+
 		}
 	}
-	if !validFederatedKey {
-		return oops.Errorf("Invalid federation key %s", keyField)
+
+	if federatedOnOneService && objName != "Query" && objName != "Mutation" {
+		// If the object is federated on one service, make sure every service with
+		// that object can parse the keys and fetch the object
+		for _, service := range serviceNames {
+			for _, typ := range serviceSchemasByName[service].Schema.Types {
+				if typ.Name == objName {
+					isRootObject := false
+					for _, introspectedField := range typ.Fields {
+						if introspectedField.Name == federationField {
+							isRootObject = true
+							break
+						}
+					}
+					if !isRootObject && federatedOnOneService {
```

Comment:
isn't federatedOnOneService here guaranteed to be true? since line 98 checks for it already?

Reply:
ah nice catch! fixed!

Reply:
I can still see it?

Reply:
Im not sure what you mean, it seems to be fixed 
![image](https://user-images.githubusercontent.com/21315389/89672821-26753480-d89a-11ea-85e6-38cb80b1352f.png)

Reply:
I am still seeing `if !isRootObject && federatedOnOneService {`, this is the commit I am looking at. https://github.com/samsarahq/thunder/pull/410/commits/015f7a5dd0009eb68f4b6d9680d073990219b76f

Reply:
It got merged in the commit after, updated!